jre_emul/Classes/NSString+JavaString.m

```diff
@@ -1005,6 +1004,7 @@ + (const J2ObjcClassInfo *)__metadata {
     { "indexOfString:fromIndex:", "I", 0x1, 45, 46, -1, -1, -1, -1 },
     { "intern", "LNSString;", 0x1, -1, -1, -1, -1, -1, -1 },
     { "isEmpty", "Z", 0x1, -1, -1, -1, -1, -1, -1 },
+    { "java_compareTo:", "I", 0x1, 33, 13, -1, -1, -1, -1 },
```

Comment:
Order doesn't matter. Most classes' method lists are short enough that a
linear comparison is fast enough, and large apps have so many classes that
hash tables would take too much memory. Thanks for checking, though.

On Mon, Sep 12, 2016 at 2:32 PM efirestone <notifications@github.com> wrote:

> In jre_emul/Classes/NSString+JavaString.m
> <https://github.com/google/j2objc/pull/787#discussion_r78458950>:
>
> > @@ -1005,6 +1004,7 @@ + (const J2ObjcClassInfo *)__metadata {
> >      { "indexOfString:fromIndex:", "I", 0x1, 45, 46, -1, -1, -1, -1 },
> >      { "intern", "LNSString;", 0x1, -1, -1, -1, -1, -1, -1 },
> >      { "isEmpty", "Z", 0x1, -1, -1, -1, -1, -1, -1 },
> > +    { "java_compareTo:", "I", 0x1, 33, 13, -1, -1, -1, -1 },
>
> Can someone more familiar with the way J2ObjC reflection works verify that
> changing the ordering here isn't going to throw any indexes off?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/google/j2objc/pull/787/files/c59c2899605da93d40a6003f32adf4c4776668c4#r78458950>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AC2kr3Ain-dty2xVTAvLdpmOjsn5ByZyks5qpcTcgaJpZM4JyOg7>
> .
>